lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -100,23 +154,28 @@ Ufe::ContextOps::Items UsdContextOps::getItems(
 {
     Ufe::ContextOps::Items items;
     if (itemPath.empty()) {
-        // Top-level items.  Variant sets and visibility.
-        if (fPrim.HasVariantSets()) {
-            items.emplace_back(
-                "Variant Sets", "Variant Sets", Ufe::ContextItem::kHasChildren);
-        }
-        auto attributes = Ufe::Attributes::attributes(sceneItem());
-        if (attributes) {
-            auto visibility =
-                std::dynamic_pointer_cast<Ufe::AttributeEnumString>(
-                    attributes->attribute(UsdGeomTokens->visibility));
-            if (visibility) {
-                auto current = visibility->get();
-                const std::string l = (current == UsdGeomTokens->invisible) ?
-                    std::string("Make Visible") : std::string("Make Invisible");
-                items.emplace_back("Toggle Visibility", l);
+        // Top-level items.  Variant sets and visibility. Do not add for gateway type node.
+        if (!fIsAGatewayType) {
```

Comment:
Lines 159-174 were indented inside this new if test. The gateway node will be the mayausd proxy shape.

Reply:
As I discussed with you, I'm still trying to figure out how we could have done this with a base class, and avoided the type-based conditional...

If we had a base class for the context ops that did not have variant sets and visibility, and a derived class that would have variant sets and visibility first, then would call the base class, would that work?  Ordering would be preserved.

Maybe I'm too attached to that idea, and maybe it's just a minor improvement, but my feeling is that it would help with separation of concerns.  The base class would take care of the proxy shape-only context ops, and the derived class would add in visibility and variant sets.  Let me know what you think.

Reply:
I agree with you, the base class approach isn't flexible enough to accomodate the ordering we'll need.  Case closed.